src/webserver/HttpBase.h

```diff
@@ -405,8 +405,27 @@ handle_request(
             responseStr = boost::json::serialize(response);
         }
 
+        auto warningFlag = false;
+        result["warning"] = boost::json::array{};
```

Comment:
You could try something like:
```cpp
boost::json::array& warning = result["warning"].as_array();
....
warning.emplace_back("Too many requests");
```

These are pretty much functionally equivalent, so feel free to leave it as is, too.

Reply:
I chose to take a slightly different approach that I believe is very clear.
```
boost::json::array warnings; // creates empty json array
...
warnings.emplace_back("Too many requests");
...
result["warning"] = warnings;
```
This way, the warnings array is not connected to the response json object until after all warnings, if any, have been added. Just need to make sure that result is a reference to response["result"] 😉